iot-hub/Samples/device/ConventionBasedOperations/Thermostat/ThermostatSample.cs

```diff
@@ -64,6 +83,53 @@ public async Task PerformOperationsAsync(CancellationToken cancellationToken)
             }
         }
 
+        private async Task GetWritablePropertiesAndHandleChangesAsync()
+        {
+            ClientProperties properties = await _deviceClient.GetClientPropertiesAsync();
+            ClientPropertyCollection writableProperties = properties.WritablePropertyRequests;
+            long serverWritablePropertiesVersion = writableProperties.Version;
+
+            // Check if the writable property version is outdated on the local side.
+            // For the purpose of this sample, we'll only check the writable property versions between local and server
+            // side without comparing the property values.
+            if (serverWritablePropertiesVersion > s_localWritablePropertiesVersion)
+            {            
+                _logger.LogDebug($"The writable property version cached on local is changing from {s_localWritablePropertiesVersion} to {serverWritablePropertiesVersion}.");
+
+                foreach (KeyValuePair<string, object> writableProperty in writableProperties)
+                {
+                    if (writableProperty.Key == "targetTemperature")
+                    {
+                        const string targetTemperatureProperty = "targetTemperature";
+
+                        if (writableProperties.TryGetValue(targetTemperatureProperty, out double targetTemperatureValue))

```

Comment:
Comparing this and L143, since it was confusing to you (and rightfully so) as to why we weren't getting a `WritableClientProperty` `out` here even though both the places you are enumerating over a `ClientPropertyCollection`, I think a comment here explaining the reasoning would be beneficial to users.
It would also help explain why the "ack" logic has been implemented twice.

Reply:
Good point! Will bring this to new PR when it is ready.